main.tf

```
@@ -123,6 +123,14 @@ resource "aws_s3_bucket" "origin" {
     }
   }
 
+  dynamic "versioning" {
+    for_each = var.versioning_enabled ? ["true"] : []
```

This can be simplified to this
```suggestion
    for_each = [var.versioning_enabled]
```